Add logging for service instances created via template

lkpot opened the pull request.

chris2511 commented on classes/s6rc.bbclass:
```diff
@@ -159,10 +159,52 @@ umask %s s6-setuidgid %s s6-log -d3 %s %s" %
+                        "notification-fd": "3",
+                        "dependencies": "mount-temp" }
+            else:
+                raise bb.parse.SkipRecipe("'%s' in S6RC_*_%s[%s] is not a valid option. Valid options are: %s" % (logmode, logname, "mode", "all-in-one, per-instance"))
```
Let's drop the "no-log" flag for templates and interpret an empty "mode" as no automatic logging service required.

chris2511 commented:
Please rebase on HEAD (eff1e66f86c1d2a06659d349eb36de1c4c5a7607), where the log size is configurable now and use the variable.

lkpot committed:
s6rc.bbclass: fix invalid no-log property

lkpot committed:
s6rc.bbclass: add logging to dynamic services

This change adds logging to dynamically created services. The log
service can be configured via "S6RC_TEMPLATES_%_log[]" or disabled by
setting "S6RC_TEMPLATES_%[no-log]".

By setting S6RC_TEMPLATES_%_log[mode] to "all-in-one" a dedicated
longrum service will be created. This service can be used to collect the
log messages of all service instances for this template in a single log.

Setting S6RC_TEMPLATES_%_log[mode] to "per-instance" creates a minimal
log service that spawns together with each service instance.
